repository/src/main/java/org/alfresco/repo/security/authority/AuthorityDAOImpl.java

```diff
@@ -1431,7 +1442,31 @@ public void setAuthorityDisplayName(String authorityName, String authorityDispla
             return;
         }
         nodeService.setProperty(ref, ContentModel.PROP_AUTHORITY_DISPLAY_NAME, authorityDisplayName);
+    }
 
+    @Override
+    public Pair<String, String> getAuthorityDisplayNameAndDescription(String authorityName)
+    {
+        NodeRef ref = getAuthorityOrNull(authorityName);
+        if (ref == null)
+        {
+            return Pair.nullPair();
+        }
+        Serializable displayName = nodeService.getProperty(ref, ContentModel.PROP_AUTHORITY_DISPLAY_NAME);
+        Serializable description = nodeService.getProperty(ref, ContentModel.PROP_DESCRIPTION);
+        return new Pair<>(DefaultTypeConverter.INSTANCE.convert(String.class, displayName), DefaultTypeConverter.INSTANCE.convert(String.class, description));
+    }
+
+    @Override
+    public void setAuthorityDisplayNameAndDescription(String authorityName, String authorityDisplayName, String description)
+    {
+        NodeRef ref = getAuthorityOrNull(authorityName);
+        if (ref == null)
+        {
+            return;
+        }
+        nodeService.setProperty(ref, ContentModel.PROP_AUTHORITY_DISPLAY_NAME, authorityDisplayName);
+        nodeService.setProperty(ref, ContentModel.PROP_DESCRIPTION, description);

```

Comment:
There's a `nodeService` method to set several properties at the same time which should be faster than this.